impala/dbapi/interface.py

```diff
@@ -164,9 +164,69 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         if exc_type is not None:
             reraise(exc_type, exc_val, exc_tb)
 
+def _replace_named_markers(operation, string_parameters):
+    """
+    Checks if there are any named markers as described in PEP 249,
+    if so replaces them from the given string_parameters.
+    Does *not* raise an error for unused parameters, or for parameters
+    that are not found.
+    """
+    # Iterate through the keywords, longest first, so partially
+    # overlapping ones don't replace the wrong value (:foo and :foobar)
+    for key in sorted(string_parameters, reverse=True):
+        operation = operation.replace(':' + key, string_parameters[key])
```

Comment:
hmm. yes, in this version that will be rather unpredictable (depening on key size it will be replaced again, or not). Which suggest it does indeed need something of a scanner instead of string.replace(). hmm.